File: Irrgarten-ruby/Irrgarten/combat_element.rb
# frozen_string_literal: true

module Irrgarten
class CombatElement

  def initialize(effect_,uses_)
    @effect = effect_
    @uses = uses_

  end

  def to_string
    "{#{@effect},#{@uses}"
  end

  def discard
    Dice.discard_element(@uses)
  end

  protected
  def produce_effect
    if (@uses > 0)
      @uses -=1
    else
      return 0
    end
    return @effect
  end

  private_class_method :new
  end
end

File: Irrgarten-Java/src/irrgarten/CombatElement.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author jhony
 */
public abstract class CombatElement {
    private float effect;
    private int uses; 
    
    CombatElement (float effect_, int uses_){ 
        effect=effect_;
        uses=uses_;
    }  
    
    protected float produceEffect (){
        
        if (uses > 0)uses--;
        else return 0;
     
        return effect;  
    }
    
    public String toString(){
       return "["+effect+","+uses+"]";  
    }
    
    boolean discard(){
        return Dice.discardElement(uses); 
    }
}
